### PR TITLE
Detect `cl.exe`'s path for compiler specs requiring a C compiler

### PR DESCRIPTION
**Pull request by HertzDevil**
Similar to Crystal itself, we use `vswhere` to detect the C compiler's full path for the few compiler specs that need to build an extra C object file. This allows files like `spec/compiler/codegen/extern_spec.cr` to be run without the MSVC developer command prompt.

### CODE REVIEW

**Comment by straight-shoota** [comment]
At this point it would probably be useful to memoize the mscv path.

**Comment by HertzDevil** [comment]
Within just this file, or in `Crystal::System::VisualStudio`?

**Comment by straight-shoota** [comment]
It should be cached globally in `Crystal::System::VisualStudio`. The MSVC path is unlikely to change during compiler or spec execution. And if it does, it's most likely not intended to affect later uses in the running program.

**Review by straight-shoota — APPROVED** [review]

File: spec/support/tempfile.cr [hunk 1]
```diff
@@ -44,14 +47,24 @@ def with_temp_executable(name, file = __FILE__)
...
+        # we won't be cross-compiling the specs binaries, so host and target
+        # bits are identical
+        bits = {{ flag?(:bits64) ? "x64" : "x86" }}
+        cl = Process.quote(msvc_path.join("bin", "Host#{bits}", bits, "cl.exe").to_s)
+      end
+
+      `#{cl} /nologo /c #{Process.quote(c_filename)} #{Process.quote("/Fo#{o_filename}")}`.should be_truthy
     {% else %}
-      `#{ENV["CC"]? || "cc"} #{Process.quote(c_filename)} -c -o #{Process.quote(o_filename)}`.should be_truthy
+      `#{Process.quote(ENV["CC"]? || "cc")} #{Process.quote(c_filename)} -c -o #{Process.quote(o_filename)}`.should be_truthy
```
Comment (oprypin) [inline spec/support/tempfile.cr:67]: I think it's invalid to quote the CC variable, I'm pretty sure I've seen it being set to a compiler **with a set of flags**

**Review by oprypin — APPROVED** [review]

**Pull request merged**